client/my-sites/comments/comment-threaded-list/index.jsx

```diff
@@ -0,0 +1,106 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import { filter, map, sortBy, take } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import CommentListHeader from 'my-sites/comments/comment-list/comment-list-header';
+import CommentNavigation from 'my-sites/comments/comment-navigation';
+import Notice from 'components/notice';
+import QuerySiteCommentsTree from 'components/data/query-site-comments-tree';
+import ThreadedComment from 'my-sites/comments/comment-threaded-list/threaded-comment';
+import { getSiteCommentsTree, isCommentsTreeInitialized } from 'state/selectors';
+
+export class CommentThreadedList extends Component {
+	static propTypes = {
+		postId: PropTypes.number,
+		siteFragment: PropTypes.string,
+		siteId: PropTypes.number,
+	};
+
+	state = {
+		commentsPerPage: 10,
+		pageLoaded: 1,
+	};
+
+	getRootComments = () => filter( this.props.commentsTree, { commentParentId: 0 } );
```

Comment:
Feels like this can be moved as a selector or at least into mapStateToProps. Eg we only need to calculate this when commentsTree is updated.

Reply:
An alternative would be to create a new endpoint or update the comments-tree endpoint to return this for us.